classes/submission/maps/Schema.php

```diff
@@ -492,176 +596,212 @@ protected function getPropertyReviewRounds(Collection $reviewRounds): array
-                        // one of the assignments have recommendOnly option set, consider it here
-                        $stage['currentUserCanRecommendOnly'] = false;
-                        foreach ($editorsStageAssignments as $editorsStageAssignment) {
-                            if ($editorsStageAssignment->userId == $user->getId() && $editorsStageAssignment->recommendOnly) {
-                                $stage['currentUserCanRecommendOnly'] = true;
-                                break;
-                            }
-                        }
-                    } else {
-                        // workaround for pkp/pkp-lib#4231, pending formal data model
-                        $stage['files'] = [
-                            'count' => 0
-                        ];
-                    }
-                    break;
+                // Get only decisions related to the relevant review round
+                if ($currentReviewRound->getId() != $decision->getData('reviewRoundId')) {
+                    continue;
+                }
 
-                    // Get revision files for editing and production stages.
-                    // Review rounds are handled separately in the review stage below.
-                case WORKFLOW_STAGE_ID_EDITING:
-                case WORKFLOW_STAGE_ID_PRODUCTION:
-                    $fileStages = [WORKFLOW_STAGE_ID_EDITING ? SubmissionFile::SUBMISSION_FILE_COPYEDIT : SubmissionFile::SUBMISSION_FILE_PROOF];
-                    // Revision files in this round.
-                    $stage['files'] = [
-                        'count' => Repo::submissionFile()->getCollector()
-                            ->filterBySubmissionIds([$submission->getId()])
-                            ->filterByFileStages($fileStages)
-                            ->getCount()
-                    ];
-                    break;
+                $reviewRecommendations->push($decision);
             }
+        }
 
+        $latestRecommendation = $reviewRecommendations->sortByDesc(
+            fn (Decision $recommendation) =>
+            strtotime($recommendation->getData('dateDecided'))
+        )->first();
+
+        if ($latestRecommendation) {
+
+            $recommendationData = [
+                'decision' => $latestRecommendation->getData('decision'),
+                'label' => Repo::decision()->getDecisionType($latestRecommendation->getData('decision'))->getRecommendationLabel(),
+            ];
 
-            $availableEditorialDecisions = $this->getAvailableEditorialDecisions($stageId, $submission);
-            $stage['availableEditorialDecisions'] = array_map(fn (DecisionType $decisionType) => ['id' => $decisionType->getDecision(), 'label' => $decisionType->getLabel()], $availableEditorialDecisions);
+            // Set recommendations for the deciding editor
+            if ($isCurrentUserDecidingEditor) {
+                $stages[$decision->getData('stageId')]['recommendations'][] = $recommendationData;
```

Comment:
I might have not explain this one clearly. Deciding editors needs to see last recommendation. But if there is multiple recommend only editors - he needs to see last from each of them. Which I suspect is not considered.

Reply:
Will do!